### PR TITLE
fix: updates AWS credential source

### PR DESCRIPTION
**Pull request by lsirac**
- Makes the region_url and url fields optional for the AWS credential source.
- Adds environment ID parsing to detect if breaking changes were introduced.

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java [hunk 1]
```diff
@@ -67,31 +69,44 @@
...
-     *
-     * <p>The `url` is the metadata server URL which is used to retrieve the AWS credentials.
+     * The source of the AWS credential. The credential source map must contain the
+     * `regional_cred_verification_url` field.
      *
      * <p>The `regional_cred_verification_url` is the regional GetCallerIdentity action URL, used to
      * determine the account ID and its roles.
+     *
+     * <p>The `environment_id` is the environment identifier, of format “aws${version}”. This is
+     * used to indicate to the library whether breaking changes were introduced to the underlying
```
Comment (elharo) [inline oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:79]: is used to indicate to the library --> indicates

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java [hunk 2]
```diff
@@ -67,31 +69,44 @@
...
-     * <p>The `region_url` is used to retrieve to targeted region.
-     *
-     * <p>The `url` is the metadata server URL which is used to retrieve the AWS credentials.
+     * The source of the AWS credential. The credential source map must contain the
+     * `regional_cred_verification_url` field.
      *
      * <p>The `regional_cred_verification_url` is the regional GetCallerIdentity action URL, used to
      * determine the account ID and its roles.
+     *
+     * <p>The `environment_id` is the environment identifier, of format “aws${version}”. This is
```
Comment (elharo) [inline oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:78]: of format --> in the format

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java [hunk 3]
```diff
@@ -67,31 +69,44 @@
...
+     * used to indicate to the library whether breaking changes were introduced to the underlying
+     * AWS implementation.
+     *
+     * <p>The `region_url` is used to retrieve to targeted region. Optional.
```
Comment (elharo) [inline oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:82]: is used to retrieve to --> identifies the

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java [hunk 4]
```diff
@@ -67,31 +69,44 @@
...
+     *
+     * <p>The `url` is the metadata server URL which is used to retrieve the AWS credentials.
```
Comment (elharo) [inline oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:84]: is the metadata server URL which is used --> locates the metadata server used

File: oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java [hunk 5]
```diff
@@ -229,7 +244,14 @@ private String getAwsRegion() throws IOException {
     if (region != null) {
       return region;
     }
-    region = retrieveResource(((AwsCredentialSource) credentialSource).regionUrl, "region");
+
+    AwsCredentialSource awsCredentialSource = (AwsCredentialSource) this.credentialSource;
+    if (awsCredentialSource.regionUrl == null || awsCredentialSource.regionUrl.isEmpty()) {
+      throw new IOException(
+          "Unable to determine the AWS region. The credential source does not contain the region url.");
```
Comment (elharo) [inline oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java:251]: URL

**Review by elharo — COMMENTED** [review]
minor suggestions; nothing substantive

**Review by chingor13 — APPROVED** [review]

**Pull request merged**